net/NetUtil.cpp

```diff
@@ -155,15 +158,16 @@ static HostEntry resolveDNS(const std::string& addressToCheck, std::vector<DNSCa
 
     // search for hit
     auto findIt = std::find_if(querycache.begin(), querycache.end(),
-                               [&addressToCheck](const auto& entry)->bool {
-                                 return entry.queryAddress == addressToCheck;
+                               [&addressToCheck, &port](const auto& entry)->bool {
+                                 return entry.queryAddress == addressToCheck &&
+                                        entry.queryPort == port;
                                });
     if (findIt != querycache.end())
         return findIt->hostEntry;
 
     // lookup and cache
-    HostEntry hostEntry(addressToCheck);
-    querycache.push_back(DNSCacheEntry{addressToCheck, hostEntry, now});
+    HostEntry hostEntry(addressToCheck, port ? port->c_str() : nullptr);
```

Comment:
This would become `!port.empty() ? port.c_str() : nullptr`. No need for optional.

Reply:
sure